src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -671,6 +682,27 @@ bool FileBasedWal::rollbackToLog(LogID id) {
     return true;
 }
 
+void FileBasedWal::cleanWAL() {
+    std::lock_guard<std::mutex> g(walFilesMutex_);
+    if (walFiles_.empty()) {
+        return;
+    }
+    auto now = time::WallClock::fastNowInSec();
+    // We skip the latest wal file because it is beging written now.
+    size_t index = 0;
+    auto it = walFiles_.begin();
+    auto size = walFiles_.size();
+    while (it != walFiles_.end()) {
+        if (index++ < size - 1 &&  (now - it->second->mtime() > policy_.ttl)) {
```

Comment:
maybe is index++ <= size-1 ?

Reply:
> maybe is index++ <= size-1 ?

Because the last wal is still active, so we can't remove it.

Reply:
> > maybe is index++ <= size-1 ?
> 
> Because the last wal is still active, so we can't remove it.

I think `index++ <= size-1` can ensures that the last wal will not be deleted.  I'm confused.

Reply:
> > > maybe is index++ <= size-1 ?
> > 
> > 
> > Because the last wal is still active, so we can't remove it.
> 
> I think `index++ <= size-1` can ensures that the last wal will not be deleted. I'm confused.

Exactly